MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/mas/OptimizationActivity.java

```diff
@@ -0,0 +1,211 @@
+
+            // Optimization API is limited to 12 coordinate sets
+            if (stops.size() == 12) {
+              Toast.makeText(OptimizationActivity.this, "Only 12 stops allowed on route", Toast.LENGTH_LONG).show();
+            } else {
+              map.addMarker(new MarkerOptions()
+                .position(new LatLng(point.getLatitude(), point.getLongitude()))
+                .title("1"));
+
+              stops.add(Position.fromCoordinates(point.getLongitude(), point.getLatitude()));
+
+              getOptimized(stops);
+            }
+          }
+        });
+      }
+    });
+  }
+
+  private void getOptimized(List<Position> coordinates) {
+
+    optimizedClient = new MapboxOptimizedTrips.Builder()
+      .setSource("first")
+      .setDestination("any")
+      .setCoordinates(coordinates)
+      .setOverview(DirectionsCriteria.OVERVIEW_FULL)
+      .setProfile(DirectionsCriteria.PROFILE_DRIVING)
+      .setAccessToken(Mapbox.getAccessToken())
+      .build();
+
+    optimizedClient.enqueueCall(new Callback<OptimizedTripsResponse>() {
+      @Override
+      public void onResponse(Call<OptimizedTripsResponse> call, Response<OptimizedTripsResponse> response) {
+        System.out.println(call.request().url().toString());
+
+        // You can get the generic HTTP info about the response
+        Log.d(TAG, "Response code: " + response.body().getTrips());
+        Log.d(TAG, "response: " + call.request().url().toString());
+
+        if (response.body() == null) {
+          Log.e(TAG, "No routes found, make sure you set the right user and access token.");
+          return;
+        } else if (response.body().getTrips().size() < 1) {
+          Log.e(TAG, "No routes found" + response.body().getTrips().size());
+          return;
+        }
+
+        // Print some info about the route
+        optimizedRoute = response.body().getTrips().get(0);
+
+        // Draw the route on the map
+        drawOptimized(optimizedRoute);
+      }
+
+      @Override
+      public void onFailure(Call<OptimizedTripsResponse> call, Throwable throwable) {
+        Log.e(TAG, "Error: " + throwable.getMessage());
+      }
+    });
+  }
+
+  private void drawOptimized(DirectionsRoute route) {
+    //remove old polyline
+    if (optimizedPolyline != null) {
+      map.removePolyline(optimizedPolyline);
+    }
+
+    // Convert LineString coordinates into LatLng[]
+    LineString lineString = LineString.fromPolyline(route.getGeometry(), PRECISION_6);
+    List<Position> coordinates = lineString.getCoordinates();
+    LatLng[] points = new LatLng[coordinates.size()];
+    for (int i = 0; i < coordinates.size(); i++) {
+      points[i] = new LatLng(
+        coordinates.get(i).getLatitude(),
+        coordinates.get(i).getLongitude());
+    }
+
+    // Draw Points on MapView
+    optimizedPolyline = map.addPolyline(new PolylineOptions()
+      .add(points)
+      .color(Color.parseColor("#23D2BE"))
```

Comment:
Changed to `.color(Color.parseColor(TEAL_COLOR))` with `private static final String TEAL_COLOR = "23D2BE";` at the top